### PR TITLE
Update FoldingScheme trait to take C1 & C2 as params

### PR DESCRIPTION
**Pull request by arnaucube**
Update FoldingScheme trait to take C1 & C2 as params which are used by the diverse folding schemes as a cycle of curves.

### CODE REVIEW

**Comment by arnaucube** [comment]
This PR points to the branch `feature/traits`. I propose that once this PR is merged with `feature/traits`, we do a PR from `feature/traits` to `main` so after that future PRs can point directly to `main`.

**Review by han0110 — APPROVED** [review]
LGTM, just a nitpick.

Also agree to merge `features/traits` to `main` after this PR is merged.

Another idea on the cycle of curves I had previously is to have a additional trait to reduce verbosity:

```rust
trait TwoChainCurve: CurveGroup {
    type Secondary: TwoChainCurve<
        BaseField = Self::ScalarField,
        ScalarField = Self::BaseField,
        Secondary = Self,
    >;
}
```

Then `FoldingScheme` only need one generic. But `C1 C2` might indeed be more straightforward.

File: src/lib.rs [hunk 1]
```diff
@@ -17,7 +24,12 @@ pub trait Transcript<F: PrimeField> {
     fn get_challenges(&mut self, n: usize) -> Vec<F>;
 }
 
-pub trait FoldingScheme<C: CurveGroup>: Clone + Debug {
+/// FoldingScheme defines trait that is implemented by the diverse folding schemes. It is defined
+/// over a cycle of curves (C1, C2), where:
+/// - C1 is the main curve, which ScalarField we use as our F for al the field operations
+/// - C2 is the auxiliary curve, which we use for the commitments, whose BaseField (for point
+/// coordinates) are in the C1::ScalarField
+pub trait FoldingScheme<C1: CurveGroup, C2: CurveGroup>: Clone + Debug {
```
[the remark this answers is not in the document]
Comment (arnaucube) [reply src/lib.rs]: Right! That's a good idea

**Comment by arnaucube** [comment]
Regarding having an additional trait for the `TwoChainCurve`, I think that for the moment we can go with the more straightforward option of simply using `C1, C2`.

Will now merge this PR (including your suggestion on constraining the swapped Scalar and Base fields), and will open a new PR from `feature/traits` to `main`.

**Pull request merged**